app/com/arpnetworking/metrics/portal/alerts/impl/FileSystemAlertRepository.java

```diff
@@ -0,0 +1,402 @@
+        _alerts = loadAlerts();
+        _isOpen.set(true);
+    }
+
+    @Override
+    public void close() {
+        assertIsOpen();
+        LOGGER.debug().setMessage("Closing FileSystemAlertRepository").log();
+        _isOpen.set(false);
+    }
+
+    @Override
+    public Optional<Alert> getAlert(final UUID identifier, final Organization organization) {
+        assertIsOpen();
+        return Optional.ofNullable(_alerts.get(identifier));
+    }
+
+    @Override
+    public AlertQuery createAlertQuery(final Organization organization) {
+        assertIsOpen();
+        return new DefaultAlertQuery(this, organization);
+    }
+
+    @Override
+    public QueryResult<Alert> queryAlerts(final AlertQuery query) {
+        assertIsOpen();
+        final Predicate<Alert> containsPredicate =
+                query.getContains()
+                        .map(c -> (Predicate<Alert>) a -> a.getDescription().contains(c))
+                        .orElse(e -> true);
+
+        final ImmutableList<Alert> alerts = _alerts.values().stream()
+                .filter(containsPredicate)
+                .skip(query.getOffset().orElse(0))
+                .limit(query.getOffset().orElse(1000))
+                .collect(ImmutableList.toImmutableList());
+
+        final long total = _alerts.values().stream()
+                .filter(containsPredicate)
+                .count();
+
+        return new DefaultQueryResult<>(alerts, total);
+    }
+
+    @Override
+    public long getAlertCount(final Organization organization) {
+        assertIsOpen();
+        return _alerts.size();
+    }
+
+    /* Unsupported mutation operations */
+
+    @Override
+    public int deleteAlert(final UUID identifier, final Organization organization) {
+        // Since we expect to use this repository just as every other AlertRepository,
+        // we should enforce the open-before-use invariant rather than immediately
+        // throwing on mutations.
+        assertIsOpen();
+        throw new UnsupportedOperationException("FilesystemAlertRepository is read-only");
+    }
+
+    @Override
+    public void addOrUpdateAlert(final Alert alert, final Organization organization) {
+        assertIsOpen();
+        throw new UnsupportedOperationException("FilesystemAlertRepository is read-only");
+    }
+
+    private ImmutableMap<UUID, Alert> loadAlerts() {
+        final AlertGroup group;
+        try (Reader reader = Files.newBufferedReader(_path)) {
+            group = _objectMapper.readValue(
+                    reader,
+                    AlertGroup.class);
+        } catch (final IOException e) {
+            throw new RuntimeException("Could not load alerts", e);
+        }
+        final ImmutableMap.Builder<UUID, Alert> mapBuilder = ImmutableMap.builder();
+        for (final SerializedAlert fsAlert : group.getAlerts()) {
+            final UUID uuid = fsAlert.getUUID().orElseGet(() -> computeUUID(fsAlert));
+
+            // TODO(cbriones):
```

Comment:
I've left this TODO and the one below in here since as mentioned in the description these are must-fix problems.